client/web/components/Flag/index.js

```diff
@@ -1,77 +1,46 @@
 // @flow
 
 import * as React from 'react';
-import { Image } from 'react-native'; // esling-disable-line
+import { Image, View } from 'react-native'; // esling-disable-line
 import styled from 'styled-components';
 import fetch from 'isomorphic-unfetch';
 import type { Response } from 'isomorphic-unfetch';
+import { graphql, createFragmentContainer } from 'react-relay';
+import QueryRenderer from '../query-renderer';
+import type { Location } from './__generated__/FlagQuery.graphql';
 
 const Image_ = styled(Image)`
   width: 32px;
   height: 32px;
 `;
+const ImageContainer = styled(View)`
+  width: 32px;
+  height: 32px;
+`;
 
-function getQueryForCode(code: string) {
-  return {
-    query: `{
-        location(id: "${code}") {
-          name
-          code
-          countryFlagURL
-        }
-      }
-    `,
-    variables: null,
-    operationName: null,
-  };
-}
-
-function checkStatus(response: Response): Response {
-  if (!response.ok) {
-    throw new Error(response);
-  }
-  return response;
-}
-
-type FlagProps = {|
-  code: string;
+type Props = {|
+  location: Location;
 |};
-type FlagState = {|
-    imageUrl: ?string;
-    retrieved: boolean;
-|};
-export default class Flag extends React.Component<FlagProps, FlagState> {
-  state = { imageUrl: undefined, retrieved: false };
+const Flag = (props: Props) => (
+  <Image_ source={{ uri: props.location.countryFlagURL }} />
+);
 
-  componentDidMount() {
-    if (!this.state.retrieved) {
-      fetch('https://graphql.kiwi.com', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'Accept': 'application/json',
-        },
-        body: JSON.stringify(getQueryForCode(this.props.code))
-      })
-      .then(checkStatus)
-      .then(response => response.json())
-      .then(response =>
-        this.setState({
-          imageUrl: response.data.location.countryFlagURL,
-          retrieved: true,
-        })
-      )
-      .catch(error => {
-        this.setState({ retrieved: true });
-      });
-    }
-  }
+const FlagView = (props: { code: string }) => (
+  <ImageContainer>
+    <QueryRenderer
+      query={graphql`
```

Comment:
Just curious. Does relay group the requests for all of the flags or will there be as many  requests as there are flags.